zokrates_analysis/src/assembly_transformer.rs

```diff
@@ -28,143 +32,135 @@ impl AssemblyTransformer {
 impl<'ast, T: Field> ResultFolder<'ast, T> for AssemblyTransformer {
     type Error = Error;
 
```

Comment:
The diff looks big but I only extracted `fold_assembly_statement` into the new folder methods for constraint